src/main/play-25/uk/gov/hmrc/cache/controller/CachingController.scala

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmrc.cache.controller
+
+import play.api.libs.json.{Json, Writes}
+import play.api.mvc.{Controller, Request, Result}
+import reactivemongo.api.commands.WriteConcern
+import uk.gov.hmrc.cache.model.Cache
+import uk.gov.hmrc.cache.repository.CacheRepositoryFactory
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait CachingController { self: Controller =>
+
+  import play.api.libs.json.JsValue
+  import play.api.libs.json.Json._
+
+  def cacheRepositoryFactory: CacheRepositoryFactory
+
+  def executionContext: ExecutionContext
```

Comment:
Can we make this `implicit` and avoid passing it around below?

Reply:
From my perspective this would pollute client's implicit scope. Moreover end client could require to have implicit execution context different then this one. 

I think in most cases in client this `def` would be override by `implicit val` passed in as constructor argument.